pkgs/tools/archivers/p7zip/default.nix

```diff
@@ -1,16 +1,16 @@
-{ stdenv, fetchFromGitHub, fetchpatch, lib, enableUnfree ? false }:
+{ lib, stdenv, fetchFromGitHub, enableUnfree ? false }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "p7zip";
-  version = "17.04";
+  version = "17.05";
 
   src = fetchFromGitHub {
-    owner  = "jinfeihan57";
-    repo   = pname;
-    rev    = "v${version}";
+    owner = "p7zip-project";
+    repo = "p7zip";
+    rev = "v${finalAttrs.version}";
     sha256 = {
-      free = "sha256-DrBuf2VPdcprHI6pMSmL7psm2ofOrUf0Oj0qwMjXzkk=";
-      unfree = "sha256-19F4hPV0nKVuFZNbOcXrcA1uW6Y3HQolaHVIYXGmh18=";
+      free = "sha256-4byb9bb3FxV9e3NnKHOwCBd8jumyzGr/SUYy6AoaY1Y=";
+      unfree = "sha256-SO1XtqGFnaXLIvGitZyJGra4O6cAbO2aiiCMfhlFpNA=";
     }.${if enableUnfree then "unfree" else "free"};
```

Comment:
This could be made simpler if only an if else.

Reply:
Very true but I would need to also change the update script which I really don't want to do ATM

If you do I'd be happy to add the change